booster/tools/rpc-client/RpcClient.hs

```diff
@@ -111,24 +121,38 @@ makeRequest ::
     Maybe Socket ->
     BS.ByteString ->
     (BS.ByteString -> m a) ->
+    [String] ->
     m (Maybe a)
-makeRequest _ _ Nothing _ _ = pure Nothing
-makeRequest time name (Just s) request handleResponse = do
+makeRequest _ _ Nothing _ _ _ = pure Nothing
+makeRequest time name (Just s) request handleResponse runOnly = do
     start <- liftIO $ getTime Monotonic
     logInfo_ $ "Sending request " <> name <> "..."
     logDebug_ $ "Request JSON: " <> BS.unpack request
-    liftIO $ sendAll s request
-    response <- liftIO readResponse
-    end <- liftIO $ getTime Monotonic
-    logInfo_ "Response received."
-    logDebug_ $ "Response JSON: " <> BS.unpack response
-    let timeStr = timeSpecs start end
-    logInfo_ $ "Round trip time for request '" <> name <> "' was " <> timeStr
-    when time $ do
-        logInfo_ $ "Saving timing for " <> name
-        liftIO $ writeFile (name <> ".time") timeStr
-    Just <$> handleResponse response
+    case decodeKoreRpc request of
+        RpcRequest r -> do
+            if null runOnly || reqToStr r `elem` runOnly
+                then do
+                    liftIO $ sendAll s request
+                    response <- liftIO readResponse
+                    end <- liftIO $ getTime Monotonic
+                    logInfo_ "Response received."
+                    logDebug_ $ "Response JSON: " <> BS.unpack response
+                    let timeStr = timeSpecs start end
+                    logInfo_ $ "Round trip time for request '" <> name <> "' was " <> timeStr
+                    when time $ do
+                        logInfo_ $ "Saving timing for " <> name
+                        liftIO $ writeFile (name <> ".time") timeStr
+                    Just <$> handleResponse response
+                else do
+                    logInfo_ $ "Only processing " <> unwords runOnly <> " requests. Skipping..."
```

Comment:
Maybe not necessary to output the whitelist every time (the user has provided it in an option). I'd rather think knowing what was there but skipped might be interesting.
```suggestion
                    logInfo_ $ "Skipping " <> Text.pack (reqToStr r) <> " request"
```